t/17-basetest.t

```diff
@@ -521,4 +521,18 @@ subtest rollback_activated_consoles => sub {
     is_deeply(\@selected_consoles, [{cmd => 'backend_select_console', testapi_console => 'last_milestone_console'}], 'last milestone console selected') or diag explain \@selected_consoles;
 };
 
+subtest search_for_expected_serial_failures => sub {
+    $bmwqemu::vars{BACKEND} = 'qemu';
+    my $mock_basetest = Test::MockModule->new('basetest');
+    $mock_basetest->mock(run => sub { die 'test failure' });
+    my $basetest = bless({
+            details => [],
+            name => 'foo',
+            category => 'category1',
+            execute_time => 42,
+    }, 'basetest');
+    $basetest->runtest();
+    is($basetest->{result}, 'fail', 'search for expected serial failures is working');
```

Comment:
I'm not sure whether that check is good enough to tell that `search_for_expected_serial_failures` has actually worked. That function will call another function to assign that result but other code paths do the same so the result will likely be `fail` anyways. I'm not sure to which level you need functions to be called but if `parse_serial_output_qemu` has already coverage it would make sense to mock it and check whether it is called.